gym/spaces/text.py

```diff
@@ -0,0 +1,90 @@
+"""Implementation of a space that represents textual strings."""
+from typing import Optional, Union, Set, SupportsIndex
+
+import numpy as np
+
+from gym.spaces.space import Space
+from gym.utils import seeding
+
+alphanumeric = set("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
+
+# TODO: Add support for:
+#  * lexicons
+#  * regular expressions
+#     - for sample() and contains()
+#  * tokenization
+
+class Text(Space[str]):
+    r"""A space representing a string comprised of characters from a given charset.
+    Example::
+        >>> # {"", "B5", "hello", ...}
+        >>> Text(5)
+        >>> # {"0", "42", "0123456789", ...}
+        >>> Text(min_length = 1,
+        ...      max_length = 10,
+        ...      charset = string.digits)
+    """
+    
+    def __init__(
+        self,
+        max_length: int,
```

Comment:
It would probably be best if `max_length` could be `-1`, and that would then bypass the check. This would help if the strings themselves could be arbitrarily long (e.g. if we're passing paragraphs of text for whatever reason), so users won't need to specify an arbitrary upper bound

Reply:
I can't imagine a situation in which a user would genuinely want the sampler to output random combinations of characters between certain length bounds. So, if we constrain sampling to a regular expression, are bounds on the length even necessary at all?

Reply:
We could remove `max_length` and leave it to the regex, but if the regex itself does not have a defined length bound, it could match an infinite pattern, and the length of the sampled string will depend on whichever internal limit the sample algorithm might have.